app/controllers/actions/CheckYourAnswersFilterImpl.scala

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.actions
+
+import controllers.routes
+import logging.Logging
+import models.UserAnswers
+import models.requests.OptionalDataRequest
+import pages.MoveCountryPage
+import play.api.mvc.{ActionFilter, Result}
+import play.api.mvc.Results.Redirect
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class CheckYourAnswersFilterImpl @Inject()(implicit val executionContext: ExecutionContext)
+  extends CheckYourAnswersFilter with Logging {
+
+  override protected def filter[A](request: OptionalDataRequest[A]): Future[Option[Result]] = {
+    val preparedForm = request.userAnswers.getOrElse(UserAnswers(request.userId)).get(MoveCountryPage) match {
+      case None => Future.successful(None)
+      case Some(value) if !value && request.uri.contains(routes.CheckYourAnswersController.onPageLoad().url) =>
```

Comment:
instead of checking if they're on CYA in the filter, can you apply the filter to only CYA instead in the controller / controller components, rather than on all controllers?

You can either write `andThen` solely on the onPageLoad/Submit in the controller, or make a new method in CC https://github.com/hmrc/one-stop-shop-exclusions-frontend/pull/8/files#diff-86bf860d21eede18f7bf840142e6e773ddd66fa24c55a3523fde92f9e64fb7e1L41

Reply:
No problem, will move but I will keep the check just incase it is used somewhere else and is fault tolerant